pkg/controller/servicebinding/secret_test.go

```diff
@@ -37,18 +35,96 @@ func TestSecretNew(t *testing.T) {
 	s := newSecret(
 		f.FakeDynClient(),
 		ns,
-		name,
+		secretName,
+		data,
+		secretOwnerReference,
 	)
 
-	t.Run("createOrUpdate", func(t *testing.T) {
-		u, err := s.createOrUpdate(data, secretOwnerReference)
+	t.Run("unique_secret_name", func(t *testing.T) {
+		hashedName1, err := name(secretName, data)
+		assert.NoError(t, err)
+		assert.Contains(t, hashedName1, secretName)
+
+		hashedName2, err := name(hashedName1, data)
+		assert.NoError(t, err)
+		assert.Equal(t, hashedName2, hashedName1)
+	})
+
+	t.Run("check secret creation", func(t *testing.T) {
```

Comment:
Still missing the scenario when two secrets are created, with equal data content, but having the keys sorted differently.